### PR TITLE
Add a generic distribution list endpoint

### LINKED ISSUE #2379: Add a generic distributions list endpoint

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #2379

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_distributions.py [hunk 1]
```diff
@@ -160,7 +161,13 @@ def test_03_fully_update(self):
                 self.do_fully_update_attr(key)
 
     @skip_if(bool, "distribution", False)
-    def test_04_delete_distribution(self):
+    def test_04_list(self):
+        """Test the generic distribution list endpoint."""
+        distributions = self.client.get(BASE_DISTRIBUTION_PATH)
+        assert self.distribution["pulp_href"] in [distro["pulp_href"] for distro in distributions]
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_distributions.py:167]: My options here for the test were to use this `CRUDPublicationDistributionTestCase` which doesn't use the bindings or to use the `ContentServePublicationDistributionTestCase` class which does use the bindings but for which doesn't really relate to this test?

**Review by bmbouter — APPROVED** [review]
Thank you!

**Review by gerrod3 — APPROVED** [review]

File: pulpcore/tests/functional/api/using_plugin/test_distributions.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/tests/functional/api/using_plugin/test_distributions.py:167]: So you are not saying that there will be no bindings generated for this endpoint, but that the test is really outdated. I can live with that.

**Review by mdellweg — APPROVED** [review]
Thank you!

File: pulpcore/tests/functional/api/using_plugin/test_distributions.py (hunk 1 above)
Comment (daviddavis) [reply pulpcore/tests/functional/api/using_plugin/test_distributions.py:167]: Yea, this test class predates the bindings and still uses the old way of calling the Pulp API.

**Pull request merged**